scripts/src/test/org/oppia/android/scripts/RegexPatternValidationCheckTest.kt

```diff
@@ -0,0 +1,81 @@
+package org.oppia.android.scripts
+
+import java.io.File
+import kotlin.test.assertEquals
+import org.oppia.android.scripts.RegexPatternValidationCheck
+import org.junit.Test
+import org.junit.Rule
+import org.junit.Before
+import org.junit.rules.TemporaryFolder
+import org.junit.rules.ExpectedException
+import org.oppia.android.scripts.ScriptResultConstants
+
+class RegexPatternValidationCheckTest {
+
+  @Rule
+  @JvmField
+  public var tempFolder: TemporaryFolder = TemporaryFolder()
+
+  @Rule
+  @JvmField
+  var thrown: ExpectedException = ExpectedException.none()
```

Comment:
Prefer using ``assertThrows`` from Truth for exceptions rather than ExpectedException. In fact, we could add ExpectedException to the list of prohibited patterns because it has one main drawback: it doesn't let you control where in the test the exception will occur. In longer tests this can result in tests unintentionally passing due to throwing a generic exception earlier than expected.

See uses of ``assertThrows`` elsewhere in the codebase for an idea on usage (we have a custom helper that we use for Kotlin typing).

Reply:
Revamped to use `assertThrows` instead of `ExceptedException`. PTAL.